src/lib/toolbar.ts

```diff
@@ -620,7 +621,22 @@ export const toolbarConfig: Record<ToolbarModeName, ToolbarMode> = {
       [
         {
           id: 'constraint-length',
-          disabled: (state) => !state.matches({ Sketch: 'SketchIdle' }),
+          disabled: (state) =>
+            !(
+              state.matches({ Sketch: 'SketchIdle' }) &&
+              state.can({
+                type: 'Constrain length',
+                data: {
+                  selection: state.context.selectionRanges,
+                  // dummy data is okay for checking if the constrain is possible
+                  length: {
+                    valueAst: createLiteral(1),
+                    valueText: '1',
+                    valueCalculated: '1',
+                  },
+                },
+              })
+            ),
```

Comment:
Different but related bug.
 
 Every other constraint in this drop down has a `state.can` for the constraint, I think it must have been removed when the length contraint event became more complicated, the other constraints don't take input data like this one does, but it really shouldn't have been removed, because the length button was enabled in the dropdown when it shouldn't have been. imo anyway.